### PR TITLE
CCC bug

### PR DESCRIPTION
**Pull request by local-minimum**
## Description

This fixes gridding issues both on backend and frontend for the CCC.
Installs javascript testing
Fixes the glyphicons ( #165 )

## Tests

Added tests for api and js (using jasmine)
Also manual testing in UI

## Original Traceback

```
scanomatic_1  |   File scanomatic_1  |     px_y, px_x = grid[:, grid.shape[1] - y, x]
"/usr/local/lib/python2.7/dist-packages/scanomatic/ui_server/calibration_api.py", line 572, in detect_colony
scanomatic_1  | IndexError: index 12 is out of bounds for axis 1 with size 12
```

### CODE REVIEW

File: scanomatic/ui_server_data/js/ccc.js [hunk 1]
```diff
@@ -1,889 +1,883 @@
...
+            $("#imProgress0").hide();
+            $("#imProgress1").hide();
+            $("#imProgress2").hide();
+            $("#imProgress3").hide();
+            $("#imProgress4").hide();
+            $("#imProgress5").hide();
+            $("#imProgress6").hide();
+            $("#divGridding").hide();
+            break;
+        case 1.1:
```
Comment (skymandr) [inline scanomatic/ui_server_data/js/ccc.js]: I think that floats in `case` constructs are potentially a very bad idea.

File: scanomatic/ui_server_data/js/ccc.js [hunk 2]
```diff
@@ -483,28 +483,15 @@ $(document)
...
-            }
-        }
-
-        function doSetGrayScaleTransformTask(scope, plateId, next) {
-            setGrayScaleTransform(scope, plateId, next);
-            //test(plateId, next);
+                setStep(2);
+                scope.Plate = plate;
+                scope.PlateNextTaskInQueue = next;
+                SetGrayScaleTransform(scope, scope.cccId, scope.CurrentImageId, scope.Plate, scope.AccessToken, setGrayScaleTransformSuccess, setGrayScaleTransformError);
```
Comment (skymandr) [inline scanomatic/ui_server_data/js/ccc.js]: This long line is very ugly.

File: scanomatic/ui_server_data/js/ccc.js [hunk 3]
```diff
@@ -1,889 +1,883 @@
...
+                .data(fixtures)
+                .enter()
+                .append("option");
+            options.attr("value", function(d) { return d; });
+            options.text(function(d) { return d; });
+            $("#" + elementName).selectedIndex = 0;
+        });
+        GetPinningFormats(function(formats) {
+            if (formats == null) {
+                alert("ERROR: There was a problem with the API while fetching pinnnig formats! ");
```
[the remark this answers is not in the document]
Comment (local-minimum) [reply scanomatic/ui_server_data/js/ccc.js]: Yes, but out of scope for this task

File: scanomatic/ui_server_data/js/ccc.js (hunk 2 above)
Comment (local-minimum) [reply scanomatic/ui_server_data/js/ccc.js]: yes, but also out of scope

**Pull request merged**